nupm/install.nu

```diff
@@ -23,107 +23,161 @@ def throw-error [
 
     log info $"installing package ($package.name)"
 
     match $package.type {
         "module" => {
-            let destination = $env.NUPM_HOME | path join $package.name
+            let mod_dir = $pkg_dir | path join $package.name
+
+            if ($mod_dir | path type) != dir {
+                throw-error ($"Module package '($package.name)' does not"
+                    + $" contain directory '($package.name)'")
+            }
+
+            let module_dir = module-dir --ensure
+            let destination = $module_dir | path join $package.name
+
+            if $force {
+                rm --recursive --force $destination
+            }
+
+            if ($destination | path type) == dir {
+                throw-error $"Package ($package.name) is already installed"
+            }
 
-            prepare-directory $destination
-            $path | copy-directory-to $destination
+            cp --recursive $mod_dir $module_dir
 
             if $package.scripts? != null {
                 log debug $"installing scripts for package ($package.name)"
-                install-scripts $path $package
+                $package.scripts | if $force {
+                    install-scripts --force $pkg_dir (script-dir --ensure)
+                } else {
+                    install-scripts $pkg_dir (script-dir --ensure)
+                }
             }
         },
         "script" => {
-            if "scripts" not-in $package {
-                let text = $"package is a script but does not have a `$.scripts` list"
-                throw-error "invalid_package_file" $text --span (metadata $path | get span)
+            log debug $"installing scripts for package ($package.name)"
+
+            $package.scripts?
+            | default [ ($pkg_dir | path join $"($package.name).nu") ]
+            | if $force {
+                install-scripts --force $pkg_dir (script-dir --ensure)
+            } else {
+                install-scripts $pkg_dir (script-dir --ensure)
             }
-
-            install-scripts $path $package
         },
         "custom" => {
-            if not ($path | path join "build.nu" | path exists) {
-                let text = $"package uses a custom install but no `build.nu` has been found"
-                throw-error "invalid_package_file" $text --span (metadata $path | get span)
+            let build_file = $pkg_dir | path join "build.nu"
+            if not ($build_file | path exists) {
+                let text = "package uses a custom install but no `build.nu` has"
+                        + " been found"
+                (throw-error
+                    "invalid_package_file"
+                    $text
+                    --span (metadata $pkg_dir | get span))
             }
 
-            nu ($path | path join 'build.nu')
+            let tmp_dir = tmp-dir build --ensure
+            cd $tmp_dir
+
+            nu $build_file ($pkg_dir | path join 'package.nuon')
+
+            cd -
```

Comment:
i think not using `cd -` is safer in such cases, what about
```nushell
do {
    cd $tmp_dir
    nu $build_file ($pkg_dir | path join 'package.nuon')
}
```
instead? :innocent:

Reply:
Good catch!

Reply:
done in https://github.com/nushell/nupm/pull/18/commits/0ae5405d10c9244f63a26b743df9373a096e1b9d